src/main/java/uk/gov/hmcts/juror/api/moj/repository/letter/court/ExcusalRefusalLetterListRepositoryImpl.java

```diff
@@ -1,79 +1,205 @@
+                QPoolRequest.poolRequest.poolNumber,
+                QJuror.juror.jurorNumber,
+                QJurorHistory.jurorHistory.dateCreated)
+            .from(QJurorPool.jurorPool)
+            .join(QJuror.juror).on(QJuror.juror.eq(QJurorPool.jurorPool.juror))
+            .join(QPoolRequest.poolRequest).on(QPoolRequest.poolRequest.eq(QJurorPool.jurorPool.pool))
+            .join(QJurorStatus.jurorStatus).on(QJurorStatus.jurorStatus.eq(QJurorPool.jurorPool.status))
+            .leftJoin(QJurorHistory.jurorHistory)
+            .on(QJurorHistory.jurorHistory.jurorNumber.eq(QJuror.juror.jurorNumber)
+                .and(QJurorHistory.jurorHistory.poolNumber.eq(QPoolRequest.poolRequest.poolNumber))
+                .and(QJurorHistory.jurorHistory.historyCode.eq(HistoryCodeMod.NON_EXCUSED_LETTER))
+                .and(QJurorHistory.jurorHistory.dateCreatedDateOnly.after(QJuror.juror.bureauTransferDate)))
+            .where(QJurorPool.jurorPool.isActive.isTrue()
+                .and(QJuror.juror.excusalRejected.eq("Y"))
+                .and(QJurorPool.jurorPool.owner.eq(owner))
+                .and(QPoolRequest.poolRequest.poolNumber.in(poolNumbers)))
+            .fetch();
+    }
+
+    private void filterEligibleLetterSearchCriteria(JPAQuery<Tuple> jpaQuery,
                                                     CourtLetterSearchCriteria courtLetterSearchCriteria) {
         if (!StringUtils.isEmpty(courtLetterSearchCriteria.jurorNumber())) {
-            jpaQuery.where(EXCUSAL_REFUSED_LETTER_LIST.jurorNumber.startsWith(courtLetterSearchCriteria.jurorNumber()));
+            jpaQuery.where(QJuror.juror.jurorNumber.startsWith(courtLetterSearchCriteria.jurorNumber()));
         }
 
         if (!StringUtils.isEmpty(courtLetterSearchCriteria.jurorName())) {
-            jpaQuery.where(EXCUSAL_REFUSED_LETTER_LIST.firstName.concat(" " + EXCUSAL_REFUSED_LETTER_LIST.lastName)
+            jpaQuery.where(QJuror.juror.firstName.concat(" ").concat(QJuror.juror.lastName)
                 .containsIgnoreCase(courtLetterSearchCriteria.jurorName()));
         }
 
         if (!StringUtils.isEmpty(courtLetterSearchCriteria.postcode())) {
             jpaQuery.where(
-                EXCUSAL_REFUSED_LETTER_LIST.postcode.trim().equalsIgnoreCase(courtLetterSearchCriteria.postcode()
-                    .trim()));
+                QJuror.juror.postcode.trim()
+                    .eq(courtLetterSearchCriteria.postcode().trim().toUpperCase()));
         }
 
         if (!StringUtils.isEmpty(courtLetterSearchCriteria.poolNumber())) {
-            jpaQuery.where(EXCUSAL_REFUSED_LETTER_LIST.poolNumber.startsWith(courtLetterSearchCriteria.poolNumber()));
-        }
-
-        if (!courtLetterSearchCriteria.includePrinted()) {
-            jpaQuery.where(EXCUSAL_REFUSED_LETTER_LIST.datePrinted.isNull());
+            jpaQuery.where(QJurorPool.jurorPool.pool.poolNumber.startsWith(courtLetterSearchCriteria.poolNumber()));
```

Comment:
Will be better to use QPoolRequest.poolRequest

Reply:
updated